realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -497,9 +508,15 @@ private void completedAsyncRealmObject(QueryUpdateTask.Result result) {
                     realmObject.onCompleted(rowPointer);
                     realmObject.notifyChangeListeners();
 
-                } else if (compare > 0) {
-                    // the caller has advanced we need to
-                    // retry against the current version of the caller if it's still empty
+                } else {
+                    if (compare > 0) {
+                        RealmLog.d("[COMPLETED_ASYNC_REALM_OBJECT "+ realmObject + "] , realm:" + HandlerController.this
+                                + " caller is more advanced. Schedule a retry on current version");
+                    } else {
```

Comment:
Although this is unlikely to happen, i still think we should throw like the left side line 524.